video_xblock/video_xblock.py

```diff
@@ -779,12 +782,21 @@ def get_enabled_transcripts(self):
         Get transcripts from different sources depending on current usage mode.
         """
         if self.threeplaymedia_streaming:
-            transcripts = list(self.fetch_available_3pm_transcripts())
+            transcripts = normalize_transcripts(list(self.fetch_available_3pm_transcripts()))
         else:
-            try:
-                transcripts = json.loads(self.transcripts) if self.transcripts else []
-                assert isinstance(transcripts, list)
-            except (ValueError, AssertionError):
-                log.exception("JSON parser can't handle 'self.transcripts' field value: {}".format(self.transcripts))
-                return []
-        return normalize_transcripts(transcripts)
+            transcripts = self.get_enabled_managed_transcripts()
+        return transcripts
+
+    def get_enabled_managed_transcripts(self):
+        """
+        Get `managed transcripts` (not directly streamed, like "3PlayMedia" ones) - e.g. "manual", "default".
+
+        :return: (list) transcripts that enabled but not directly streamed
+        """
+        try:
+            transcripts = json.loads(self.transcripts) if self.transcripts else []
+            assert isinstance(transcripts, list)
```

Comment:
This is the best place to catch possible dirty (not "json-deserializable") transcripts values if any will occasionally appear in xblock's String field (and they will appear in a future because of js code base design).